### PR TITLE
SPCTR-2479-Fix: backend font loading issue in footer

### PR DESCRIPTION
**Pull request by chandan-30**
### Description
This PR resolves the issue with the font not loading in the backend of the footer. The problem was caused by a missing Google Fonts URL, which resulted in the browser blocking the font due to mixed content. To fix this issue, we have updated the 'content_pre_render' function in the class-uagb-rest-api.php file to include a <link> tag with the Google Fonts URL of the selected font family.

### Checklist:
- [x] My code is tested
- [x] My code follows accessibility standards 
- [x] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [ ] I've added proper labels to this pull request

### CODE REVIEW

**Review by aapatre — APPROVED** [review]

File: classes/class-uagb-rest-api.php
Comment (RajkiranBagal) [inline classes/class-uagb-rest-api.php:1]: I see only this file change is needed in this PR, Why other files are changed? Please undo other file changes.

File: classes/class-uagb-rest-api.php [hunk 1]
```diff
@@ -88,8 +94,19 @@ public function content_pre_render( $block ) {
 			}
 
 			$block_css_style = $desktop_css . $tab_styling_css . $mob_styling_css;
+			$style           = ! empty( $block_css_style ) ? '<style class="uagb-widgets-style-renderer">' . $block_css_style . '</style>' : '';
+
+			if ( 'uagb/advanced-heading' === $block['blockName'] && ! empty( $style ) ) {
```
Comment (RajkiranBagal) [inline classes/class-uagb-rest-api.php:99]: Is the issue only for the Heading block? Do other blocks work fine?

File: classes/class-uagb-rest-api.php
Comment (chandan-30) [reply classes/class-uagb-rest-api.php:1]: Made changes. Removed all the other files

File: classes/class-uagb-rest-api.php (hunk 1 above)
Comment (chandan-30) [reply classes/class-uagb-rest-api.php]: I have rechecked all the blocks and found few blocks are having same issue, so I changed code logic which works for all.

**Review by RajkiranBagal — APPROVED** [review]

File: classes/class-uagb-rest-api.php [hunk 2]
```diff
@@ -88,8 +91,25 @@ public function content_pre_render( $block ) {
...
+					}
+				}
+				$style = $style . $link_tag_list;
+			}
+
 
-			$style = ! empty( $block_css_style ) ? '<style class="uagb-widgets-style-renderer">' . $block_css_style . '</style>' : '';
 			array_push( $block['innerContent'], $style );
 			return $block;
 		}
```
Comment (rahulvarma722) [inline classes/class-uagb-rest-api.php:115]: Please verify once and modify your code

`
function content_pre_render( $block ) {
	$tab_styling_css  = '';
	$mob_styling_css  = '';
	$UAGB_Post_Assets = new UAGB_Post_Assets( get_the_ID() );

	$assets = $UAGB_Post_Assets->get_block_css_and_js( $block );

	$desktop_css = isset( $assets['css']['desktop'] ) ? $assets['css']['desktop'] : '';

	if ( isset( $assets['css']['tablet'] ) && ! empty( $assets['css']['tablet'] ) ) {
		$tab_styling_css .= '@media only screen and (max-width: ' . UAGB_TABLET_BREAKPOINT . 'px) {';
		$tab_styling_css .= $assets['css']['tablet'];
		$tab_styling_css .= '}';
	}

	if ( isset( $assets['css']['mobile'] ) && ! empty( $assets['css']['mobile'] ) ) {
		$mob_styling_css .= '@media only screen and (max-width: ' . UAGB_MOBILE_BREAKPOINT . 'px) {';
		$mob_styling_css .= $assets['css']['mobile'];
		$mob_styling_css .= '}';
	}

	$block_css_style = $desktop_css . $tab_styling_css . $mob_styling_css;

	if( empty( $block_css_style ) ){
		return $block;
	}

	$link_tag_list = '';
	// This line of code creates a new array named $font_family_attrs by searching through the keys of an existing array.
	$font_family_attrs = preg_grep( '/fontfamily/i', array_keys( $block['attrs'] ) );

	foreach ( $font_family_attrs as $attr ) {
		if ( ! empty( $block['attrs'][ $attr ] ) ) {
			// Get the font family value and construct the Google Fonts URL.
			$gfont_url   = 'https://fonts.googleapis.com/css?family=' . urlencode( $block['attrs'][ $attr ] );

			// Create a link tag for the stylesheet with the constructed URL.
			$link_tag_list .= '<link rel="stylesheet" href="' . esc_url( $gfont_url ) . '" media="all">';
		}
	}

	$style = '<style class="uagb-widgets-style-renderer">' . $block_css_style . '</style>';
	$style = $style . $link_tag_list;
	array_push( $block['innerContent'], $style );

	return $block;
}`

**Review by rahulvarma722 — CHANGES_REQUESTED** [review]

File: classes/class-uagb-rest-api.php (hunk 2 above)
Comment (chandan-30) [reply classes/class-uagb-rest-api.php:112]: @rahulvarma722 Made required changes, Please re-review.

File: classes/class-uagb-rest-api.php [hunk 3]
```diff
@@ -68,28 +68,45 @@ public function content_pre_render( $block ) {
...
 			}
 
 			$block_css_style = $desktop_css . $tab_styling_css . $mob_styling_css;
 
-			$style = ! empty( $block_css_style ) ? '<style class="uagb-widgets-style-renderer">' . $block_css_style . '</style>' : '';
+			if ( empty( $block_css_style ) ) {
+				return $block;
+			}
+
+			$style = '<style class="uagb-widgets-style-renderer">' . $block_css_style . '</style>';
```
Comment (rahulvarma722) [inline classes/class-uagb-rest-api.php:94]: This should be in before line number 108 and after 107

File: classes/class-uagb-rest-api.php [hunk 4]
```diff
@@ -68,28 +68,45 @@ public function content_pre_render( $block ) {
...
+
+			$link_tag_list = '';
+			foreach ( $font_family_attrs as $attr ) {
+				if ( ! empty( $block['attrs'][ $attr ] ) ) {
+					// Get the font family value and construct the Google Fonts URL.
+					$gfont_url = 'https://fonts.googleapis.com/css?family=' . urlencode( $block['attrs'][ $attr ] );
+					// Create a link tag for the stylesheet with the constructed URL.
+					$link_tag_list .= '<link rel="stylesheet" href="' . esc_url( $gfont_url ) . '" media="all">';
+				}
+			}
```
Comment (rahulvarma722) [inline classes/class-uagb-rest-api.php:107]: Leave a space after this statement

**Review by rahulvarma722 — CHANGES_REQUESTED** [review]

File: classes/class-uagb-rest-api.php (hunk 4 above)
Comment (chandan-30) [reply classes/class-uagb-rest-api.php:105]: It's done.

**Review by rahulvarma722 — APPROVED** [review]

**Review by RajkiranBagal — APPROVED** [review]

**Review by RajkiranBagal — APPROVED** [review]

**Pull request merged**